### PR TITLE
HexEditor: Add highlight annotations

### PR DESCRIPTION
**Pull request by AtkinsSJ**
[Peek 2024-01-10 19-39 hexeditor annotations.webm](https://github.com/SerenityOS/serenity/assets/222642/c607f2e8-2889-486b-9e13-16961f181d59)

This is an initial implementation of annotations. They currently are limited to a background colour, and they aren't saved anywhere. I already have a bit of a todo list for making them better but wanted to get the first pass submitted.

I forgot to record it in the video, but there's also an "Add Annotation" button in the main menu, and the context menu works equally in the hex and text areas.

## Plans
- List annotations in a GUI panel
- Save and load them from a file (separate from the file you're editing)
- Allow adding text to them, so you can actually label things as well as highlight them

### CODE REVIEW

File: Userland/Applications/HexEditor/HexDocument.h [hunk 1]
```diff
@@ -14,9 +15,18 @@
 #include <AK/WeakPtr.h>
 #include <LibCore/Forward.h>
 #include <LibGUI/Command.h>
+#include <LibGfx/Color.h>
 
 constexpr Duration COMMAND_COMMIT_TIME = Duration::from_milliseconds(400);
 
+struct Annotation {
+    size_t start_offset;
+    size_t end_offset;
```
Comment (LucasChollet) [inline Userland/Applications/HexEditor/HexDocument.h:24]: Very nit! It would be nice to zero-initialize that.

**Review by LucasChollet — APPROVED** [review]
Very neat!
I haven't played with it yet, but I'm definitely looking forwards more `HexEditor` enhancements.

**Comment by AtkinsSJ** [comment]
**Changes:**
- Initialise Annotation fields
- Add icons by @cubiclove - please let me know if you want the `co-authored-by` email changed. :^) (And please nobody merge before then!)

**Comment by cubiclove** [comment]
No thanks, looks good to me!

File: Userland/Applications/HexEditor/EditAnnotationDialog.cpp [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    m_background_color = find_descendant_of_type_named<GUI::ColorInput>("background_color");
+    m_save_button = find_descendant_of_type_named<GUI::DialogButton>("save_button");
+    m_cancel_button = find_descendant_of_type_named<GUI::DialogButton>("cancel_button");
+
+    // FIXME: This could be specified in GML, but the GML doesn't like property setters that aren't `set_FOO()`.
+    m_background_color->set_color_has_alpha_channel(false);
+
+    // NOTE: The NumericInput stores an i64, so not all size_t values can fit. But I don't think we'll be
+    //       hex-editing files larger than 9000 petabytes for the foreseeable future!
+    VERIFY(hex_document.size() <= INT64_MAX);
```
Comment (DanShaders) [inline Userland/Applications/HexEditor/EditAnnotationDialog.cpp:61]: Totally a nitpick, but in the light of #21995, can we do `NumericLimits<i64>::max()` here and further down the file?

Comment (AtkinsSJ) [reply Userland/Applications/HexEditor/EditAnnotationDialog.cpp]: Sure! I couldn't remember how to do that properly when I wrote this.

**Pull request merged**